Mage.Sets/src/mage/cards/o/OmnathLocusOfAll.java

```diff
@@ -0,0 +1,139 @@
+        super(effect);
+    }
+
+    @Override
+    public OmnathLocusOfAllManaEffect copy() {
+        return new OmnathLocusOfAllManaEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player != null) {
+            player.getManaPool().setManaBecomesBlack(true);
+        }
+        return true;
+    }
+}
+
+class OmnathLocusOfAllCardEffect extends OneShotEffect {
+    public OmnathLocusOfAllCardEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "look at the top card of your library. You may reveal that card if it has three or more " +
+                "colored mana symbols in its mana cost. If you do, add three mana in any combination of colors and " +
+                "put it into your hand. If you don't reveal it, put it into your hand.";
+    }
+
+    public OmnathLocusOfAllCardEffect(final OmnathLocusOfAllCardEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public OmnathLocusOfAllCardEffect copy() {
+        return new OmnathLocusOfAllCardEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+        Card card = player.getLibrary().getFromTop(game);
+        if (card == null) {
+            return false;
+        }
+        player.lookAtCards(null, card, game);
+        int amountOfDifferentManaColors = 0;
+        List<ColoredManaSymbol> manaList = new ArrayList<>();
+        if (card.getManaCost().containsColor(ColoredManaSymbol.W)) {
+            amountOfDifferentManaColors++;
+            manaList.add(ColoredManaSymbol.W);
+        }
+        if (card.getManaCost().containsColor(ColoredManaSymbol.U)) {
+            amountOfDifferentManaColors++;
+            manaList.add(ColoredManaSymbol.U);
+        }
+        if (card.getManaCost().containsColor(ColoredManaSymbol.B)) {
+            amountOfDifferentManaColors++;
+            manaList.add(ColoredManaSymbol.B);
+        }
+        if (card.getManaCost().containsColor(ColoredManaSymbol.R)) {
+            amountOfDifferentManaColors++;
+            manaList.add(ColoredManaSymbol.R);
+        }
+        if (card.getManaCost().containsColor(ColoredManaSymbol.G)) {
+            amountOfDifferentManaColors++;
+            manaList.add(ColoredManaSymbol.G);
+        }
+        if (amountOfDifferentManaColors > 2) {
+            if (!player.chooseUse(outcome, "Reveal " + card.getName() + '?', source, game)) {
+                player.moveCards(card, Zone.HAND, source, game);
+                return true;
+            }
+            player.revealCards(source, new CardsImpl(card), game);
+            new AddManaInAnyCombinationEffect(3, manaList.toArray(new ColoredManaSymbol[0])).apply(game, source);
```

Comment:
I'm pretty sure the colors you add are any colors of the card and not just its mana cost, so you should use the color of the card itself.